plugins/processors/starlark/starlark.go

```diff
@@ -40,6 +41,46 @@ type Starlark struct {
 	applyFunc *starlark.Function
 	args      starlark.Tuple
 	results   []telegraf.Metric
+	cache     Cache
+}
+
+// The struct that defines the shared cache used
+type Cache struct {
```

Comment:
this should be called `State` rather than `Cache`. Cache implies it can be flushed or dropped with no ill effects. In this case that's not necessarily true.

Reply:
@ssoroka make senses